### PR TITLE
Improve workflow event triggers

### LINKED ISSUE #22958: action cannot be triggered when using on.pull_request
### Description

Action cannot be triggered when using `on.pull_request`.

New pr can trigger this action.

But push to pr source branch cannot trigger it.

Github can trigger it.

```yaml
# .gitea/workflows/build.yaml
name: Gitea Actions Demo
run-name: ${{ github.actor }} is testing out Gitea Actions 🚀
on:
  pull_request:
    types: [opened, edited, synchronize, reopened]
jobs:
  Explore-Gitea-Actions:
    runs-on: ubuntu-latest
    steps:
      - run: echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
      - run: echo "🐧 This job is now running on a ${{ runner.os }} server hosted by Gitea!"
      - run: echo "🔎 The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
      - name: Check out repository code
        uses: actions/checkout@v3
      - run: echo "💡 The ${{ github.repository }} repository has been cloned to the runner."
      - run: echo "🖥️ The workflow is now ready to test your code on the runner."
      - name: List files in the repository
        run: |
          ls ${{ github.workspace }}
      - run: echo "🍏 This job's status is ${{ job.status }}."
```

### Gitea Version

ca445c

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build Gitea myself
run from command-line

### Database

SQLite

### LINKED ISSUE #23674: Action status check in pr is not updated
### Description

Action status check is not updated on the pull_request synchronize event.

```yml
# .gitea/workflows/build.yaml
name: Gitea Actions Demo
run-name: ${{ github.actor }} is testing out Gitea Actions 🚀
on:
  pull_request: yes
  push:
    branches:
      - main
jobs:
  Explore-Gitea-Actions:
    runs-on: ubuntu-20.04
    steps:
      - run: echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
      - run: echo "🐧 This job is now running on a ${{ runner.os }} server hosted by Gitea!"
      - run: echo "🔎 The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
      - name: Check out repository code
        uses: actions/checkout@v3
      - run: echo "💡 The ${{ github.repository }} repository has been cloned to the runner."
      - run: echo "🖥️ The workflow is now ready to test your code on the runner."
      - name: List files in the repository
        run: |
          ls ${{ github.workspace }}
      - run: echo "🍏 This job's status is ${{ job.status }}."
```

I create a PR `#3` from `dev` to `main`, and the status check is enabled. When I push a commit `1703af8060` to `dev`, action is triggered. But the status check on the PR page is not updated.

commit page
image

action page
image

### Gitea Version

df411819eb

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build Gitea myself
run from command-line

### Database

SQLite

### PR DESCRIPTION
**Pull request by Zettat123**
Follow #23037 
Fix [#22598 comment](https://github.com/go-gitea/gitea/issues/22958#issuecomment-1475763042)

Workflows with `pull_request` trigger event can't be triggered by `pull_request_sync` event. This PR adds the `canGithubEventMatch` function to check if a Github event can match any Gitea event. If the Github event matches a Gitea event, the related workflows should be triggered.

### CODE REVIEW

**Comment by delvh** [comment]
Wrong fixing issue?

**Comment by Zettat123** [comment]
> Wrong fixing issue?

This PR is to fix a bug mentioned in https://github.com/go-gitea/gitea/issues/22958#issuecomment-1475763042. I've updated the  description.

**Review by lunny — APPROVED** [review]

File: modules/actions/github.go [hunk 1]
```diff
@@ -25,17 +25,48 @@ const (
 	githubEventPullRequestComment       = "pull_request_comment"
 )
 
-func convertFromGithubEvent(evt *jobparser.Event) string {
+// canGithubEventMatch check if the input Github event can match any Gitea event.
+func canGithubEventMatch(evt *jobparser.Event, triggedEvent webhook_module.HookEventType) bool {
```
Comment (wolfogre) [inline modules/actions/github.go]: Is it possible to redefine it like:
```suggestion
func canGithubEventMatch(eventName string, triggedEvent webhook_module.HookEventType) bool {
```

And could you please add some unit tests for it?

Comment (Zettat123) [reply modules/actions/github.go]: Of course. I will add some tests.

**Comment by Zettat123** [comment]
> And could you please add some unit tests for it?

I've added some unit tests in the latest commit, please review again

**Review by wolfogre — APPROVED** [review]

**Pull request merged**